typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -1,33 +1,56 @@
 import { ethers, providers } from 'ethers';
 
 import {
-  ERC20__factory,
+  HypERC20CollateralVaultDeposit__factory,
   HypERC20Collateral__factory,
+  HypERC20__factory,
 } from '@hyperlane-xyz/core';
-import { ERC20Metadata, ERC20RouterConfig } from '@hyperlane-xyz/sdk';
-import { Address } from '@hyperlane-xyz/utils';
+import {
+  ERC20Metadata,
+  IsmType,
+  TokenRouterConfig,
+  TokenType,
+} from '@hyperlane-xyz/sdk';
+import { Address, isZeroishAddress, rootLogger } from '@hyperlane-xyz/utils';
 
 import { DEFAULT_CONTRACT_READ_CONCURRENCY } from '../consts/concurrency.js';
 import { EvmHookReader } from '../hook/EvmHookReader.js';
-import { EvmIsmReader } from '../ism/EvmIsmReader.js';
+import {
+  DerivedIsmConfigWithAddress,
+  EvmIsmReader,
+} from '../ism/EvmIsmReader.js';
 import { MultiProvider } from '../providers/MultiProvider.js';
-import { ChainName } from '../types.js';
+import { ChainNameOrId } from '../types.js';
+
+type WarpRouteBaseMetadata = Record<'mailbox' | 'owner' | 'hook', string> & {
+  interchainSecurityModule?: DerivedIsmConfigWithAddress;
+};
 
-type WarpRouteBaseMetadata = Record<
-  'mailbox' | 'owner' | 'token' | 'hook' | 'interchainSecurityModule',
-  string
+/**
+ * @remark
+ * We only expect to support deriving a subset of these types, for now.
+ */
+export type DerivedTokenType = Extract<
+  TokenType,
+  'collateral' | 'collateralVault' | 'native' | 'synthetic'
 >;
 
-type DerivedERC20WarpRouteConfig = Omit<ERC20RouterConfig, 'type' | 'gas'>;
+export type DerivedTokenRouterConfig = TokenRouterConfig & {
+  type: DerivedTokenType;
+  interchainSecurityModule?: DerivedIsmConfigWithAddress;
+};
```

Comment:
not the biggest fan of these subsets/separate types but I think my PR on main already cleaned this up a bit

Reply:
Removed